state/git/impl.go

```diff
@@ -0,0 +1,176 @@
+// Copyright © 2017 The Kubicorn Authors
```

Comment:
This looks solid to me, but I would like to better understand the workflow here.
The committing part should work as expected, when you change cluster, it should update the repository as well.

But I'm wondering can we improve workflow for reading as well. I'm not sure what @kris-nova wanted initially, but I'm wondering would it be useful to check remote on read as well. For example, if you want to work with a cluster but it doesn't exist on your local machine, should we check remote and if it exists there to pull it?

What do you think about this idea? Would it be useful and possible to implement something similar?

Reply:
What you say here makes sense, thought of it at first. I'll fix this while at the Airport, need to catch a flight now.

Reply:
@xmudrii that's a great idea. Fixed, there seems to be a merge conflict though, don't know why for sure.